source/common/config/README.md

```diff
@@ -0,0 +1,47 @@
+# xDS
```

Comment:
I'm a bit worried how this file here get attention from developer / updated when things changed. We don't have good structure of developer docs currently though.  @envoyproxy/maintainers thoughts?